src/OrchardCore/OrchardCore.Sms.Core/Services/SmsProviderFactory.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using OrchardCore.Entities;
+using OrchardCore.Settings;
+
+namespace OrchardCore.Sms.Services;
+
+public class SmsProviderFactory : ISmsProviderFactory
```

Comment:
Simplify

Reply:
@sebastienros I removed the factory and simplified it. Let me know your thoughts

https://github.com/OrchardCMS/OrchardCore/pull/14125/files#diff-cc9687e2af5bb116baccc4fab6fe03cb0d9947eef620be03fc44cc5e865f71faR15-R33